torch/csrc/utils/python_arg_parser.h

```diff
@@ -218,6 +223,22 @@ inline std::vector<int64_t> PythonArgs::intlistWithDefault(int i, std::vector<in
   return res;
 }
 
+static at::Type& default_type() {
+  if (!THPDefaultATenType) {
+    throw std::runtime_error("THPDefaultATenType not initialized");
+  }
+  return *torch::autograd::VariableType::getType(*THPDefaultATenType);
+}
+
+inline const at::Type& PythonArgs::dtype(int i) {
+  if (!args[i]) return default_type();
+  PyObject *arg = args[i];
+  if (!THPDtype_Check(arg)) {
```

Comment:
Don't double-check the argument type. It's already checked by the parser.

Reply:
sure; fwiw this seems to be done often in these functions (see e.g. Generator).